tutorials/users/remote-processing/remote-processing.md

```diff
@@ -1,7 +1,199 @@
+      packages](https://pavics-weaver.readthedocs.io/en/latest/package.html#application-package) and deploy them for 
+      later execution.
+- [ESGF-CWT](https://pavics-weaver.readthedocs.io/en/latest/processes.html#esgf-cwt)
+  - These processes are executed by a remote server through the [ESGF Compute API](https://github.com/ESGF/esgf-compute-api)
+    interface.
+
+```{note}
+Weaver converts all process types to OGC API - Processes internally. This allows users to interact with all processes
+using the same REST interface that Weaver provides. 
+
+This makes dealing with multiple different process types easier since the user can interact with all process types 
+through the same interface.
+
+For example, WPS 1/2 processes normally have a [SOAP](https://en.wikipedia.org/wiki/SOAP) interface. Weaver simplifies
+the interaction for the user by translating the SOAP payloads to JSON as expected by Weaver's REST interface. In other
+words, if a user can use Weaver, they don't have to learn the WPS protocol in order to execute WPS processes.
+```
+
+### Workflow Processes
+
+By chaining atomic processes into multistep workflows, Weaver can be used to accomplish complex data processing tasks.
+
+Weaver dispatches each step in the workflow to the appropriate node in the network for execution. It handles data
+transfers if needed and tries to minimize data transfers by executing workflow steps on the same node where the data
+needed for that step is hosted. Only the result from the last step in the process will be returned to the user.
+
+Workflows are defined using the [common workflow language (CWL)](https://www.commonwl.org/) which describes which atomic
+processes to execute in what order and how the data should be transferred between steps. Workflows can contain atomic
+processes hosted on different nodes in the network if needed.
+
+## The Weaver python client
+
+The Weaver software comes with a command line interface (CLI) and a python client. In order to run either, you need the
+weaver python package installed. 
+
+This package may already be installed if you're working in a [Marble Jupyterlab](../ide/jupyterlab-howto.md) 
+environment. To test whether the package is installed, run the following command in a JupyterLab console:
+
+```
+import weaver
+```
+
+If the package is not installed you can install it from the git repository with the `pip` command. Run the following
+in a JupyterLab console:
+
+```
+import sys
+import subprocess
+subprocess.run([sys.executable, "-m", "pip", "install", "git+https://github.com/crim-ca/weaver.git"])
+```
+
+### Using the python client
+
+The main component of Weaver is the process and so most of the interaction that you will have with the Weaver API will
+involve inspecting, executing, and managing processes.
+
+To interact with Weaver, first create a `weaver_client` object:
+
+```
+from weaver.cli import WeaverClient
+weaver_client = WeaverClient("https://<your-node-url>/weaver")
+```
```

Comment:
Do you think giving an example of a full weaver url would be helpful to the user?

Reply:
No, I'm really trying to avoid those

Reply:
Ok.